sw/device/lib/dif/README.md

```diff
@@ -0,0 +1,7 @@
+# the OpenTitan DIF Library
+
+A DIF is a "Device Interface Function".
```

Comment:
Perhaps also a quick one-liner indicating how this is different (or the same) as hardware device drivers for the less SW-inclined folks.

Reply:
Hopefully I have covered this point in the two added sentences. If not, I would welcome suggestions about how to improve the phrasing.